src/dataset.ts

```diff
@@ -341,6 +347,49 @@ export function checkStructure(s: JsObject): string {
         }
     }
 
+    if ('shape' in s) {
+        if (s['shape'] !== undefined) {
+            if (typeof s['shape'] === 'object' && s['shape'] !== null) {
```

Comment:
this will silently accept non-objects
```suggestion
            if (typeof s['shape'] !== 'object' || s['shape'] === null) {
                return "'shape' must be an object";
            }
```